wpilibc/src/main/native/cpp/event/BooleanEvent.cpp

```diff
@@ -7,42 +7,48 @@
 using namespace frc;
 
 BooleanEvent::BooleanEvent(EventLoop* loop, std::function<bool()> condition)
-    : m_loop(loop), m_condition(std::move(condition)) {}
+    : m_loop(loop),
+      m_condition(std::make_shared<std::function<bool()>>(condition)) {
+  m_state = std::make_shared<bool>((*m_condition)());
+  m_loop->Bind(
+      // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
+      [condition = *m_condition, state = m_state] { *state = condition(); });
```

Comment:
```suggestion
      [condition = m_condition, state = m_state] { *state = (*condition)(); });
```
This should avoid unnecessarily copying the `std::function`, which should be more efficient since it avoids allocating (and later deallocating) memory.